core/valid_pow.go

```diff
@@ -0,0 +1,36 @@
+package xchaincore
+
+import (
+	"github.com/xuperchain/xuperunion/common/log"
+	"github.com/xuperchain/xuperunion/consensus"
+	"github.com/xuperchain/xuperunion/ledger"
+	"github.com/xuperchain/xuperunion/pb"
+)
+
+func ValidPowBlock(block *pb.Block, xcore *XChainCore) bool {
+	internalBlock := block.GetBlock()
+	if xcore == nil || block == nil || internalBlock == nil {
+		log.Warn("invalid: xcore or block or internalBlock is nil")
+		return false
+	}
+
+	// validation for consensus of pow, if ok, tell the miner to stop mining
+	newBlockHeight := internalBlock.GetHeight()
+	if xcore.con.Type(newBlockHeight) == consensus.ConsensusTypePow {
+		if newBlockHeight < xcore.Ledger.GetMeta().GetTrunkHeight() {
+			log.Warn("invalid block: new block's height is not enough")
```

Comment:
it whould be better for trouble shooting, if blocks'height & trunk height are printed here.